src/main/java/com/checkmarx/flow/service/ResultsService.java

```diff
@@ -191,12 +191,50 @@ private void handleJiraCase(ScanRequest request, ScanResults results) throws Jir
             log.info("Processing results with JIRA issue tracking");
             jiraService.process(results, request);
         } catch (RestClientException e) {
+            Map<String, ScanResults.XIssue> nonPublishedScanResultsMap = jiraService.getNonPublishedScanResults();
             if (e.getStatusCode().isPresent() && e.getStatusCode().get() ==  HttpStatus.NOT_FOUND.value()) {
                 throw new JiraClientRunTimeException("Jira service is not accessible for URL: " + jiraService.getJiraURI(), e);
+            } else if (e.getStatusCode().isPresent() &&
+                    e.getStatusCode().get() ==  HttpStatus.BAD_REQUEST.value() &&
+                    nonPublishedScanResultsMap.size() > 0) {
+                throwExceptionWhenPublishingErrorOccurred(e, nonPublishedScanResultsMap);
+
             } else {
                 throw  e;
             }
+        } catch (JiraClientException e) {
+            Map<String, ScanResults.XIssue> nonPublishedScanResultsMap = jiraService.getNonPublishedScanResults();
+            if (nonPublishedScanResultsMap.size() > 0) {
+                throwExceptionWhenPublishingErrorOccurred(e, nonPublishedScanResultsMap);
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    private void throwExceptionWhenPublishingErrorOccurred(Exception e, Map<String, ScanResults.XIssue> nonPublishedScanResultsMap) {
+        throw new JiraClientRunTimeException("Wasn't been able to publish the next issues into JIRA:\n"
+                + printNonPublishedScanResults(nonPublishedScanResultsMap) + "\nwith the following reason: " + e.getMessage());
+    }
+
+    private String printNonPublishedScanResults(Map<String, ScanResults.XIssue> nonPublishedScanResultsMap) {
```

Comment:
I would add some unit tests for this method (not necessary Cucumber, a Junit is fine)